### ISSUE TITLE
Reps and weight are 0 when entering plan page

### ISSUE THREAD

**Issue by dmrowietz** [brandonp2412/Flexify#123]
When entering the page where the exercises are listed and sets can be input, the reps and weight are initially set to 0. Tapping on the exercise will update reps and weight to the correct numbers.

1. Go to "Plans" Tab
2. Click on any plan
3. In the newly opened page, reps and weight are 0, as described above

Expected behavior:
reps and weight should be set to their proper values when first entering the page.

I am encountering the bug on Linux native in version 1.1.87. With version 1.1.86 I do not have this bug.

My guess: Has something to do with [this commit](https://github.com/brandonp2412/Flexify/commit/0264a07d90dc4e0bbaaa020f7faf46f5cc8d81ea)

**Comment by brandonp2412** [comment]
Not sure how that commit could cause this, if you could please upload more details to try and reproduce this bug. 

Perhaps the `readTable` method in drift isn't reliable on all devices?

ADB logs or flexify app db dump would be appreciated.

**Comment by dmrowietz** [comment]
One important piece of information that I forgot to write yesterday: This bug only occurs in 1.1.87 and later. 1.1.86 works fine. That's why I am suspecting that particular commit in the first place.

I have dug a little deeper into this. To pinpoint the problem I did the following in a test environment: 

1. In Branch https://github.com/dmrowietz/Flexify/tree/issue_123_debugging I removed all but one item from defaultExercises and defaultPlans to make the DB as simple as possible. I then built up the db-query step by step:
a: join
b: orderBy
c: groupBy
d: addColumns
![image](https://github.com/user-attachments/assets/896a51d5-670a-424c-837a-10111ecf04ef)

3. Delete existing flexify.sqlite file <-- make sure this is the db of the test environment, not your actual database
4. Build flexify on the branch issue_123_debugging. I am building on Linux native, building for Android is broken for me at the moment.
5. Start app and add one set of "Arnold press" (1rep@1kg), restart the app
6. The DB is now very simple. It contains only one exercise with exactly one set with 1rep@1kg
7. Using a debugger take a look at the variables a,b,c,d in updateDefaults()
![image](https://github.com/user-attachments/assets/b347f33a-1fad-4de8-bb08-4240b9739b8c)
=> Curious, groupBy reduces the number of items. In my understanding it should not do that.
More details:
![image](https://github.com/user-attachments/assets/80a76a12-c35a-4d67-ba39-cbd88f7add41)
b._parsedData[1] is the set I created with 1rep@1kg
Only _parsedData[0] gets past groupBy so i think this is where the problem lies.
